omod-common/src/main/java/org/openmrs/module/webservices/rest/web/resource/impl/BaseDelegatingResource.java

```diff
@@ -263,6 +263,22 @@ public T newInstance(String type) {
 	 */
 	protected abstract void delete(T delegate, String reason, RequestContext context) throws ResponseException;
 	
+	/**
+	 * Unvoid or unretire delegate, whichever action is appropriate for the resource type.
+	 * Subclasses need to override this method, which is called internally by
+	 * {@link #undelete(String, RequestContext)}.
+	 * 
+	 * @param delegate
+	 * @param context
+	 * @throws ResponseException
+	 * @return Object
+	 */
+	protected T undelete(T delegate, RequestContext context) throws ResponseException {
+		//Default implimentation of this method if not overriden by sub-class is to raise an 
```

Comment:
Can you fix this typo "implimentation"?

Reply:
Thank you @samuelmale , I've got this resolved.